lib/asyn-ares.c

```diff
@@ -361,11 +372,13 @@ CURLcode Curl_resolver_is_resolved(struct connectdata *conn,
 /*
  * Curl_resolver_wait_resolv()
  *
- * waits for a resolve to finish. This function should be avoided since using
+ * Waits for a resolve to finish. This function should be avoided since using
  * this risk getting the multi interface to "hang".
  *
  * If 'entry' is non-NULL, make it point to the resolved dns entry
  *
+ * This should only be called when 'conn' is in CURLM_STATE_WAITRESOLVE.
```

Comment:
I tried to document the existing calling requirement here.  I didn't attempt to change it.